streams/src/main/java/org/apache/kafka/streams/state/ValueAndTimestamp.java

```diff
@@ -29,8 +29,8 @@
     private final V value;
     private final long timestamp;
 
-    private ValueAndTimestamp(final V value,
-                              final long timestamp) {
+    protected ValueAndTimestamp(final V value,
```

Comment:
Since the only usage of the constructor is in this class, it should be ok to be private.

Also, since it's just a data struct, we could also make the class final.